src/Rules/ValidYoutubeVideo.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Alaouy\Youtube\Rules;
+
+use Illuminate\Contracts\Validation\Rule;
+use Alaouy\Youtube\Facades\Youtube;
+
+class ValidYoutubeVideo implements Rule
+{
+    /**
+     * Create a new rule instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        //
+    }
+
+    /**
+     * Determine if the validation rule passes.
+     *
+     * @param  string  $attribute
+     * @param  mixed  $value
+     * @return bool
+     */
+    public function passes($attribute, $value)
+    {
+        try {
+            $videoId = Youtube::parseVidFromURL($value);
+            $video = Youtube::getVideoInfo($videoId);
```

Comment:
You should define the $part argument as ['id'] (cost 0 quota) as you only need to check, not fetch the actual video data.

Reply:
Sure thing, will change this